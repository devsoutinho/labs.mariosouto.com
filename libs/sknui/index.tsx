export { Box } from "./src/components/Box/web";
export { Text } from "./src/components/Text/web";