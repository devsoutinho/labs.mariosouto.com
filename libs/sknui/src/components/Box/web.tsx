import React from "react";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface BoxProps {
  tag?: string;
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  // receive anything
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}
export function Box({ tag, styleSheet, children, ...props }: BoxProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const Tag = tag || ("div" as any);
  return (
    <Tag
      style={{
        display: "flex",
        flexDirection: "column",
        ...styleSheet,
      }}
      {...props}
    >
      {children}
    </Tag>
  );
}
