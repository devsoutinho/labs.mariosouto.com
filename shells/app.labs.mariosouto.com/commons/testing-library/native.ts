import { render as rtlRender } from "@testing-library/react-native";
export * from "@testing-library/react-native";

interface DefaultWrapperProps {
  children: React.ReactNode;
}
function DefaultWrapper({ children }: DefaultWrapperProps) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return (ui: any, options = {}) => {
    return rtlRender(ui, {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      wrapper: AllTheProviders as any,
      ...options,
    });
  };
};

export const render = () =>
  console.error("Please create your own render method using customRender()");
