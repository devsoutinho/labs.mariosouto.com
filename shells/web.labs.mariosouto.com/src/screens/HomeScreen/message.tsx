export const message = "OlÃ¡ pessoas!";
