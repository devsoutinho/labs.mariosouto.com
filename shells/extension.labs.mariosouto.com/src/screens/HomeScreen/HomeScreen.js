import React from "react";
import { Box, Text } from "@devsoutinho/sknui";
export function HomeScreen() {
  return React.createElement(
    Box,
    {
      styleSheet: {
        flex: 1,
        backgroundColor: "#000",
        alignItems: "center",
        justifyContent: "center",
      },
    },
    React.createElement(
      Text,
      {
        styleSheet: {
          color: "#fff",
        },
      },
      "Ol\u00E1 pessoas!"
    )
  );
}
