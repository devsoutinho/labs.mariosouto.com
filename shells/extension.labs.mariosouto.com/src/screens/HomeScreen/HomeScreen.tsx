import React from "react";
import { Box, Text } from "@devsoutinho/sknui/web";

export function HomeScreen() {
  const [message, setMessage] = React.useState("");
  React.useEffect(() => {
    chrome.storage.local.get(["wordCountApp"], (result) => {
      const { length, totalWords } = result.wordCountApp || {};
      setMessage(
        `Selected text length: ${length} | Selected text total words: ${totalWords}`
      );
    });
  }, []);

  return (
    <Box
      styleSheet={{
        width: "320px",
        flex: 1,
        backgroundColor: "#000",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Text
        styleSheet={{
          color: "#fff",
        }}
      >
        {message}
      </Text>
    </Box>
  );
}
