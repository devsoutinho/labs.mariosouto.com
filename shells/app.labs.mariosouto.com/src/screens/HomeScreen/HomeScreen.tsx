import React from "react";
import { Box, Text, Touchable, useTheme } from "@devsoutinho/sknui/native";
import { message } from "@src/screens/HomeScreen/message";
import { StatusBar } from "expo-status-bar";

export function HomeScreen({ navigation }: RootStackHomeScreenProps) {
  const { theme } = useTheme();

  return (
    <>
      <StatusBar style="auto" />
      <Box
        styleSheet={{
          flex: 1,
          backgroundColor: {
            xs: theme.colors.primary.x100,
          },
          alignItems: "center",
          justifyContent: "center",
        }}
      >
        <Text
          styleSheet={{
            color: "#000",
          }}
        >
          {message}
        </Text>
        <Touchable onTap={() => navigation.navigate("storybook")}>
          <Text
            styleSheet={{
              color: "#000",
            }}
          >
            Go to Storybook
          </Text>
        </Touchable>
      </Box>
    </>
  );
}
