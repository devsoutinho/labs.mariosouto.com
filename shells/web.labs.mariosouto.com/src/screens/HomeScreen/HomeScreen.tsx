import React from "react";
import { Box, Text, Form, InputText, Button, Icon } from "@devsoutinho/sknui/web";
import { message } from "@src/screens/HomeScreen/message";

export function HomeScreen() {
  return (
    <Box
      styleSheet={{
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Form onSubmit={() => {
        window.location.href = "/dashboard";
      }}>
        <InputText label="Email"  />

        <Button
          type="submit"
          suffix={<Icon name="arrow_left" />}
        >
          Entrar
        </Button>
        <Box
          styleSheet={{
            marginTop: "16px",
          }}
        >
          <Text>
            Ou fa√ßa login com
          </Text>
          <Box>
            <Button>
              Github
            </Button>
          </Box>
        </Box>
      </Form>
    </Box>
  );
}
