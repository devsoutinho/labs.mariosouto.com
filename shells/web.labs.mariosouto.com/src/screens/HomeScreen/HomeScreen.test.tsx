import React from "react";
import { HomeScreen } from "./HomeScreen";
import { customRender } from "@devsoutinho/common-test/testing-library/web";
import { Provider } from "@devsoutinho/sknui/web";

const render = customRender((props) => <Provider {...props} />);

describe("<HomeScreen />", () => {
  it("renders as expected", () => {
    const { container } = render(<HomeScreen />);

    expect(container).toMatchSnapshot();
  });
});
