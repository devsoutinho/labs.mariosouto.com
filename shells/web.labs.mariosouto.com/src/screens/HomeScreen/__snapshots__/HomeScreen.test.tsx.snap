// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders as expected 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-0 backgroundColor {
  xs: #FFE8D9;
  sm: #FFD0B5;
  md: #FFB088;
  lg: #FF9466;
  xl: #F9703E;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 16px;
}

<div>
  <div
    class="emotion-0"
  >
    <form
      class="emotion-1"
    >
      <label>
        Email
        <input
          style="display: flex;"
          type="text"
        />
      </label>
      <button
        style="background: transparent; display: flex; gap: 8px;"
      >
        Entrar agora!
        <div
          class="emotion-1"
        >
          arrow_left
        </div>
      </button>
      <div
        class="emotion-3"
      >
        <p>
          Ou fa√ßa login com
        </p>
        <div
          class="emotion-1"
        >
          <button
            style="background: transparent; display: flex; gap: 8px;"
          >
            Github
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
